### PR TITLE
Issues/#26

### PR DESCRIPTION
**Pull request by kimuray**
企業情報編集画面を作成しました。
また企業マイページから情報編集ページへ遷移するリンクを作成しました。

### CODE REVIEW

File: app/controllers/companies_controller.rb [hunk 1]
```diff
@@ -1,4 +1,74 @@
...
+  end
+
+  # POST /companies
+  # POST /companies.json
+  def create
+    @company = Company.new(company_params)
+
+    respond_to do |format|
+      if @company.save
+        format.html { redirect_to @company, notice: 'Company was successfully created.' }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.html 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: app/controllers/companies_controller.rb [hunk 2]
```diff
@@ -1,4 +1,74 @@
...
+        format.json { render :show, status: :created, location: @company }
+      else
+        format.html { render :new }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.html 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: app/controllers/companies_controller.rb [hunk 3]
```diff
@@ -1,4 +1,74 @@
...
+
+  # POST /companies
+  # POST /companies.json
+  def create
+    @company = Company.new(company_params)
+
+    respond_to do |format|
+      if @company.save
+        format.html { redirect_to @company, notice: 'Company was successfully created.' }
+        format.json { render :show, status: :created, location: @company }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.json 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: app/views/companies/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<h1>Editing Company</h1>
+<%= @user.company.name %>
```
Comment (mineko-fuji) [inline app/views/companies/edit.html.erb]: __[rails_best_practices]__
* law of demeter [:ref](http://rails-bestpractices.com/posts/2010/07/24/the-law-of-demeter/)

File: test/controllers/companies_controller_test.rb [hunk 5]
```diff
@@ -1,9 +1,49 @@
...
+    @company = companies(:one)
+  end
+
   test "should get index" do
     get :index
     assert_response :success
+    assert_not_nil assigns(:companies)
+  end
+
+  test "should get new" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/views/companies/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+<%= form_for(@company) do |f| %>
```
Comment (mineko-fuji) [inline app/views/companies/_form.html.erb]: __[rails_best_practices]__
* replace instance variable with local variable [:ref](http://rails-bestpractices.com/posts/2010/07/24/replace-instance-variable-with-local-variable/)

File: app/controllers/companies_controller.rb [hunk 7]
```diff
@@ -1,4 +1,74 @@
...
+      else
+        format.html { render :new }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.json 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: test/controllers/companies_controller_test.rb [hunk 8]
```diff
@@ -1,9 +1,49 @@
...
+    get :new
+    assert_response :success
+  end
+
+  test "should create company" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/views/companies/_form.html.erb [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  <% if @company.errors.any? %>
```
Comment (mineko-fuji) [inline app/views/companies/_form.html.erb]: __[rails_best_practices]__
* replace instance variable with local variable [:ref](http://rails-bestpractices.com/posts/2010/07/24/replace-instance-variable-with-local-variable/)

File: app/controllers/companies_controller.rb [hunk 10]
```diff
@@ -1,4 +1,74 @@
...
+    @company = Company.new
+  end
+
+  # GET /companies/1/edit
+  def edit
+  end
+
+  # POST /companies
+  # POST /companies.json
+  def create
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [TooManyStatements] has approx 10 statements [:ref](https://github.com/troessner/reek/blob/master/docs/Too-Many-Statements.md)

File: test/controllers/companies_controller_test.rb [hunk 11]
```diff
@@ -1,9 +1,49 @@
...
+    assert_difference('Company.count') do
+      post :create, company: {  }
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Unnecessary spacing detected.
* Space inside empty hash literal braces detected. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: app/views/companies/_form.html.erb [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+    <div id="error_explanation">
+      <h2><%= pluralize(@company.errors.count, "error") %> prohibited this company from being saved:</h2>
```
Comment (mineko-fuji) [inline app/views/companies/_form.html.erb]: __[rails_best_practices]__
* replace instance variable with local variable [:ref](http://rails-bestpractices.com/posts/2010/07/24/replace-instance-variable-with-local-variable/)

File: app/controllers/companies_controller.rb [hunk 13]
```diff
@@ -1,4 +1,74 @@
...
+      else
+        format.html { render :edit }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /companies/1
+  # DELETE /companies/1.json
+  def destroy
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [TooManyStatements] has approx 6 statements [:ref](https://github.com/troessner/reek/blob/master/docs/Too-Many-Statements.md)

File: test/controllers/companies_controller_test.rb [hunk 14]
```diff
@@ -1,9 +1,49 @@
...
+    end
+
+    assert_redirected_to company_path(assigns(:company))
+  end
+
+  test "should show company" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/views/companies/_form.html.erb [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+
+      <ul>
+      <% @company.errors.full_messages.each do |message| %>
```
Comment (mineko-fuji) [inline app/views/companies/_form.html.erb]: __[rails_best_practices]__
* replace instance variable with local variable [:ref](http://rails-bestpractices.com/posts/2010/07/24/replace-instance-variable-with-local-variable/)

File: app/controllers/companies_controller.rb [hunk 16]
```diff
@@ -1,4 +1,74 @@
...
+      end
+    end
+  end
+
+  # PATCH/PUT /companies/1
+  # PATCH/PUT /companies/1.json
+  def update
+    respond_to do |format|
+      if @company.update(company_params)
+        format.html { redirect_to @company, notice: 'Company was successfully updated.' }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.html 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: test/controllers/companies_controller_test.rb [hunk 17]
```diff
@@ -1,9 +1,49 @@
...
+    get :show, id: @company
+    assert_response :success
+  end
+
+  test "should get edit" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/controllers/companies_controller.rb [hunk 18]
```diff
@@ -1,4 +1,74 @@
...
+        format.json { render :show, status: :ok, location: @company }
+      else
+        format.html { render :edit }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.html 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: test/controllers/companies_controller_test.rb [hunk 19]
```diff
@@ -1,9 +1,49 @@
...
+    get :edit, id: @company
+    assert_response :success
   end
 
+  test "should update company" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/controllers/companies_controller.rb [hunk 20]
```diff
@@ -1,4 +1,74 @@
...
+    end
+  end
+
+  # PATCH/PUT /companies/1
+  # PATCH/PUT /companies/1.json
+  def update
+    respond_to do |format|
+      if @company.update(company_params)
+        format.html { redirect_to @company, notice: 'Company was successfully updated.' }
+        format.json { render :show, status: :ok, location: @company }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.json 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: test/controllers/companies_controller_test.rb [hunk 21]
```diff
@@ -1,9 +1,49 @@
...
+    patch :update, id: @company, company: {  }
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Unnecessary spacing detected.
* Space inside empty hash literal braces detected. [:ref](https://github.com/bbatsov/ruby-style-guide#spaces-operators)

File: app/controllers/companies_controller.rb [hunk 22]
```diff
@@ -1,4 +1,74 @@
...
+      else
+        format.html { render :edit }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [DuplicateMethodCall] calls format.json 2 times [:ref](https://github.com/troessner/reek/blob/master/docs/Duplicate-Method-Call.md)

File: test/controllers/companies_controller_test.rb [hunk 23]
```diff
@@ -1,9 +1,49 @@
...
+    assert_redirected_to company_path(assigns(:company))
+  end
+
+  test "should destroy company" do
```
Comment (mineko-fuji) [inline test/controllers/companies_controller_test.rb]: __[rubocop]__
* Prefer single-quoted strings when you don't need string interpolation or special symbols. [:ref](https://github.com/bbatsov/ruby-style-guide#consistent-string-literals)

File: app/controllers/companies_controller.rb [hunk 24]
```diff
@@ -1,4 +1,74 @@
...
+      else
+        format.html { render :new }
+        format.json { render json: @company.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /companies/1
+  # PATCH/PUT /companies/1.json
+  def update
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[reek]__
* [TooManyStatements] has approx 9 statements [:ref](https://github.com/troessner/reek/blob/master/docs/Too-Many-Statements.md)

File: app/models/user.rb [hunk 25]
```diff
@@ -1,6 +1,8 @@
 class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
+  belongs_to :company
+
```
Comment (mineko-fuji) [inline app/models/user.rb]: __[rubocop]__
* Trailing whitespace detected. [:ref](https://github.com/bbatsov/ruby-style-guide#no-trailing-whitespace)

File: app/models/company.rb [hunk 26]
```diff
@@ -0,0 +1,3 @@
+class Company < ActiveRecord::Base
```
Comment (mineko-fuji) [inline app/models/company.rb]: __[rubocop]__
* Missing top-level class documentation comment.

File: app/controllers/companies_controller.rb (hunk 1 above)
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [89/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb (hunk 7 above)
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [83/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb (hunk 16 above)
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [89/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb (hunk 22 above)
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [83/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb [hunk 27]
```diff
@@ -1,4 +1,74 @@
...
+      end
+    end
+  end
+
+  # DELETE /companies/1
+  # DELETE /companies/1.json
+  def destroy
+    @company.destroy
+    respond_to do |format|
+      format.html { redirect_to companies_url, notice: 'Company was successfully destroyed.' }
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [94/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb [hunk 28]
```diff
@@ -1,4 +1,74 @@
...
+      format.json { head :no_content }
+    end
+  end
+
+  private
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Keep a blank line before and after `private`.

File: app/controllers/companies_controller.rb [hunk 29]
```diff
@@ -1,4 +1,74 @@
...
+    # Use callbacks to share common setup or constraints between actions.
+    def set_company
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Inconsistent indentation detected.

File: app/controllers/companies_controller.rb [hunk 30]
```diff
@@ -1,4 +1,74 @@
...
+      @company = Company.find(current_user.company_id)
+      @user = User.find(current_user.id)
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [88/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

File: app/controllers/companies_controller.rb [hunk 31]
```diff
@@ -1,4 +1,74 @@
...
+    def company_params
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Inconsistent indentation detected.

File: app/controllers/companies_controller.rb [hunk 32]
```diff
@@ -1,4 +1,74 @@
...
+      params.require(:company).permit(:email, :name, :dev_content, :tel, :address)
```
Comment (mineko-fuji) [inline app/controllers/companies_controller.rb]: __[rubocop]__
* Line is too long. [82/80] [:ref](https://github.com/bbatsov/ruby-style-guide#80-character-limits)

**Comment by lasershow** [comment]
@kimuray has_oneの件承知しました。
おそらく担当者は複数人登録できた方が、企業側も楽だと思うので。書き換えましょう！

**Comment by kimuray** [comment]
@lasershow has_oneからhas_manyに変更しましたー。

**Pull request merged**